src/overmap.cpp

```diff
@@ -4021,46 +3939,49 @@ int overmap::random_special_rotation( const overmap_special &special, const trip
         if( valid && score >= top_score ) {
             if( score > top_score ) {
                 top_score = score;
-                rotations.clear();  // New top score. Forget previous rotations.
+                last = first; // New top score. Forget previous rotations.
             }
-            rotations.push_back( r );
+            *last++ = r;
```

Comment:
Expressions like that are confusing. Breaking it into 2 lines would improve readability.